### PR TITLE
Tests of GMG DC Picard Stokes

### PR DESCRIPTION
**Pull request by zjiaqi2018**
2 tests of the GMG iterated defect correction Stokes solver. 
 @MFraters   GMG+DC Picard converges in these two cases. 
I cannot reproduce the case in https://github.com/geodynamics/aspect/issues/4563
because the world builder won't work.

### CODE REVIEW

File: tests/nonlinear_channel_flow_tractions_dc_picard_Stokes_GMG.prm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#Test GMG  iterated defect correction Stokes solver based on
+# benchmarks/newton_solver_benchmark_set/nonlinear_channel_flow/input_t.prm
+
+include $ASPECT_SOURCE_DIR/benchmarks/newton_solver_benchmark_set/nonlinear_channel_flow/input_t.prm
+set Additional shared libraries = $ASPECT_SOURCE_DIR/benchmarks/newton_solver_benchmark_set/nonlinear_channel_flow/libsimple_nonlinear.so
```
Comment (tjhei) [inline tests/nonlinear_channel_flow_tractions_dc_picard_Stokes_GMG.prm:5]: Remove this line. Instead, you need to add a .cc with the same name as the prm that contains an #include of the cc file. Take a look at similar tests.

File: tests/tosi_benchmark_gmg_dc_picard_solver.prm [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Test GMG  iterated defect correction Stokes solver based on
+# benchmarks/newton_solver_benchmark_set/tosi_et_al_2015/input.prm
+
+set Additional shared libraries            = $ASPECT_SOURCE_DIR/benchmarks/tosi_et_al_2015_gcubed/libtosi_benchmark.so
```
Comment (tjhei) [inline tests/tosi_benchmark_gmg_dc_picard_solver.prm:4]: Same here.

**Review by tjhei — COMMENTED** [review]
Thank you. Can you confirm that statistics and visual output is similar to not using GMG?

**Comment by zjiaqi2018** [comment]
nonlinear channel: identical
image
left: GMG+ DC Picard, right: AMG
image
image

**Comment by zjiaqi2018** [comment]
Tosi benchmark using "single Advection, iterated defect correction Stokes", timestep 126:
image
image
image
image

File: tests/nonlinear_channel_flow_tractions_dc_picard_Stokes_GMG.prm [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#Test GMG  iterated defect correction Stokes solver based on
+# benchmarks/newton_solver_benchmark_set/nonlinear_channel_flow/input_t.prm
+
+set Additional shared libraries = $ASPECT_SOURCE_DIR/benchmarks/newton_solver_benchmark_set/nonlinear_channel_flow/libsimple_nonlinear.so
```
Comment (tjhei) [inline tests/nonlinear_channel_flow_tractions_dc_picard_Stokes_GMG.prm:4]: Please remove this line.

**Comment by tjhei** [comment]
Great, you will need to update the test output.

**Comment by zjiaqi2018** [comment]
> Great, you will need to update the test output.

Why is the test diff file empty? 
image

**Comment by tjhei** [comment]
Use https://jenkins.tjhei.info/blue/organizations/jenkins/aspect/detail/PR-4896/4/artifacts for now. We might need to fix the other tester.

**Review by tjhei — APPROVED** [review]

**Pull request merged**